### PR TITLE
break(framework) Disable `flower-server-app`

### PR DESCRIPTION
**Pull request by chongshenng**
Merge after 
* #4491 

Merge before 
* #4485

### CODE REVIEW

File: e2e/test_reconnection.sh [hunk 1]
```diff
@@ -22,67 +22,116 @@ case "$1" in
...
-timeout 2m flower-supernode ./ --insecure $rest_arg --superlink $server_address &
+timeout 10m flower-supernode ./ --insecure $rest_arg --superlink $server_address \
+  --isolation="subprocess" --clientappio-api-address="localhost:9095" &
 cl2_pid=$!
 echo "Starting second client"
 sleep 3
 
 # Kill superlink, this should send the clients into their retry loops
-kill $sl_pid
+check_and_kill "$sl_pids"
```
Comment (panh99) [inline e2e/test_reconnection.sh:67]: Is `sl_pids` set somewhere?

File: e2e/test_reconnection.sh [hunk 2]
```diff
@@ -22,67 +22,116 @@ case "$1" in
...
-fi
+# Check for "Success" in a loop with a timeout
+while [ "$found_success" = false ] && [ $elapsed -lt $timeout ]; do
+    if grep -q "Run finished" flwr_output.log; then
+        echo "Training worked correctly!"
+        found_success=true
+        kill $cl1_pid; kill $cl2_pid
+        sleep 3
+        check_and_kill "$cl1_pids"
+        check_and_kill "$sl_pids"
```
Comment (panh99) [inline e2e/test_reconnection.sh:122]: Just a q question, where do we set `cl1_pids` and `sl_pids`?

File: e2e/test_reconnection.sh (hunk 1 above)
Comment (chongshenng) [reply e2e/test_reconnection.sh:67]: Yes, for `flower-superlink`: 
```
sl_pids=$(pgrep -f "flower-superlink")
```

File: e2e/test_reconnection.sh (hunk 2 above)
Comment (chongshenng) [reply e2e/test_reconnection.sh:122]: You're right, `cl1_pids` is not set. I'll remove it.

File: e2e/test_reconnection.sh [hunk 3]
```diff
@@ -22,67 +22,114 @@ case "$1" in
...
+pip install -e . --no-deps
+
+# Remove any duplicates
+sed -i '/^\[tool\.flwr\.federations\.e2e\]/,/^$/d' pyproject.toml
+
+# Append the federations config to pyproject.toml
+echo -e $"\n[tool.flwr.federations.e2e]\naddress = \"127.0.0.1:9093\"\ninsecure = true" >> pyproject.toml
+sleep 1
+
+timeout 10m flower-superlink --insecure $db_arg $rest_arg --executor flwr.superexec.deployment:executor &
```
Comment (danieljanes) [inline e2e/test_reconnection.sh:49]:
```suggestion
timeout 10m flower-superlink --insecure $db_arg $rest_arg &
```

Comment (chongshenng) [reply e2e/test_reconnection.sh:49]: Will fix the next one too.

**Pull request merged**